core/src/main/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProvider.scala

```diff
@@ -110,6 +110,25 @@ class LocalEntityProvider(workspace: Workspace, implicit protected val dataSourc
     }
   }
 
+  override def deleteEntitiesOfType(entityType: String): Future[Int] = {
+    dataSource.inTransaction { dataAccess =>
+      traceDBIO("LocalEntityProvider.deleteEntitiesOfType") { rootSpan =>
+        rootSpan.putAttribute("workspaceId", OpenCensusAttributeValue.stringAttributeValue(workspaceContext.workspaceId))
+        rootSpan.putAttribute("entityType", OpenCensusAttributeValue.stringAttributeValue(entityType))
+
+        dataAccess.entityQuery.getActiveIdsForType(workspaceContext.workspaceIdAsUUID, entityType) flatMap { idsForType =>
+          dataAccess.entityQuery.countReferringEntities(workspace, idsForType.keys.toSet) flatMap { referringEntities =>
+            if (referringEntities != 0)
+              throw new DeleteEntitiesOfTypeConflictException(referringEntities)
+            else {
+              dataAccess.entityQuery.hide(workspaceContext, idsForType.values.toSeq)
```

Comment:
to do this right - and be efficient about user requests for deleting large tables - we need a new `hide()` method. The current `hide()` method accepts a potentially quite long list of ids and uses `WHERE … IN (many ids)`, which is super inefficient for this use case. Instead it should use a `WHERE entity_type='whatever' and deleted = 0`, which would be much cleaner.

Reply:
Addressed in f1011c1650d87c6502526a89de774cfef3df3544